warranty/models/sale_order.py

```diff
@@ -0,0 +1,14 @@
+from odoo import models, fields, api
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    has_warranty_product = fields.Boolean(compute="_compute_has_warranty_product", default=False)
+
+    @api.depends("order_line.product_id")
```

Comment:
```suggestion
    @api.depends("order_line")
```

I believe this should be sufficient

Reply:
yeah, done!